src/main/java/org/alliancegenome/curation_api/services/helpers/diseaseAnnotations/DiseaseAnnotationCurieManager.java

```diff
@@ -5,29 +5,29 @@
 
 public abstract class DiseaseAnnotationCurieManager {
 
-	static Map<String, DiseaseAnnotationCurie> curieTaxonMap = new HashMap<>();
+	static Map<String, DiseaseAnnotationCurie> curieSpeciesNameMap = new HashMap<>();
 
 	static {
-		curieTaxonMap.put("NCBITaxon:7955", new ZFINDiseaseAnnotationCurie());
-		curieTaxonMap.put("NCBITaxon:9606", new RGDDiseaseAnnotationCurie());
-		curieTaxonMap.put("NCBITaxon:10116", new RGDDiseaseAnnotationCurie());
-		curieTaxonMap.put("NCBITaxon:10090", new MGIDiseaseAnnotationCurie());
-		curieTaxonMap.put("NCBITaxon:6239", new WormDiseaseAnnotationCurie());
-		curieTaxonMap.put("NCBITaxon:7227", new FlyDiseaseAnnotationCurie());
-		curieTaxonMap.put("NCBITaxon:559292", new SGDDiseaseAnnotationCurie());
+		curieSpeciesNameMap.put("Danio rerio", new ZFINDiseaseAnnotationCurie());
+		curieSpeciesNameMap.put("Homo sapiens", new RGDDiseaseAnnotationCurie());
+		curieSpeciesNameMap.put("Rattus norvegicus", new RGDDiseaseAnnotationCurie());
+		curieSpeciesNameMap.put("Mus musculus", new MGIDiseaseAnnotationCurie());
+		curieSpeciesNameMap.put("Caenorhabditis elegans", new WormDiseaseAnnotationCurie());
+		curieSpeciesNameMap.put("Drosophila melanogaster", new FlyDiseaseAnnotationCurie());
+		curieSpeciesNameMap.put("Saccharomyces cerevisiae", new SGDDiseaseAnnotationCurie());
```

Comment:
This could be using the ENUM also.

Reply:
This is how it was implemented with taxon IDs - any reason to change?

Reply:
Doesn't mean the implementation was correct at the time. I was under the impression that the endpoints were going to go away.

Reply:
Sure, I don’t mind changing it if there is a reason that an enum would be better. Just wondering what that reason is?

Reply:
@oblodgett Given you can't have spaces in an enum constant, wouldn't it be less efficient to have to do a string conversion and then check the enum rather than just using a map?  Or is there some advantage of an enum that I'm not aware of?

Reply:
This way the strings, ids are in 1 spot. Also you can do the lookup in the enum. This is what we do in the:
https://github.com/alliance-genome/agr_java_software/blob/master/agr_java_core/src/main/java/org/alliancegenome/neo4j/entity/SpeciesType.java

And there is helper methods to do the lookups and conversions.

Reply:
But the strings and IDs are not a 1-to-1 relationship.  I think you are suggesting that the enum constants would be the species names (converted to a form that is valid for an enum constant), but there are multiple taxon IDs for Saccharomyces cerevisiae, for example.

Reply:
you can put a list in an enum and do that lookup that way.

Reply:
But do we really want to list out all the NCBITaxon IDs for Saccharomyces cerevisiae?  There are 37 in our database, and probably a whole lot more that just haven't been loaded into our database (yet).

Reply:
What about using a vocabularySet? Im trying to avoid hardcoded string in the code. If this code has then some other code has it, then we have hardcoded lists of string all over the code and when we need to "add another one" it makes that task and all the testing very difficult

Reply:
We could just lose the CANONICAL_TAXONS list in the ValidationConstants.  The only place it is used is the genomic entity validation to check the taxon ID - if it's not found then it compares species name, so the taxon ID check isn't really necessary I just wrote it like that as in the majority of cases the taxon ID will match and we don't have to do any processing of the species name.  That's probably an insignificant gain though, so if we lose that then we are only ever doing a lookup on species name, in which case an enum makes sense.